components/datadog/apps/nginx/k8s/modifiers.go

```diff
@@ -0,0 +1,164 @@
+// on a DeploymentArgs struct and returns PodSpecArgs
+func ensureDeploymentPodTemplateSpec(d *appsv1.DeploymentArgs) (*corev1.PodSpecArgs, bool) {
+	// nil check spec
+	deploymentSpecPtr := d.Spec
+	if deploymentSpecPtr == nil {
+		d.Spec = &appsv1.DeploymentSpecArgs{}
+	}
+	// type check spec
+	spec, ok := deploymentSpecPtr.(*appsv1.DeploymentSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check spec.Template
+	podTemplatePtr := spec.Template
+	if podTemplatePtr == nil {
+		spec.Template = &corev1.PodTemplateSpecArgs{}
+	}
+	// type check spec.Template
+	podTemplate, ok := podTemplatePtr.(*corev1.PodTemplateSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check spec.Template.Spec
+	podTemplateSpecPtr := podTemplate.Spec
+	if podTemplateSpecPtr == nil {
+		podTemplate.Spec = &corev1.PodSpecArgs{}
+	}
+
+	// type check spec.Template.Spec
+	podTemplateSpec, ok := podTemplateSpecPtr.(*corev1.PodSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	return podTemplateSpec, true
+}
+
+// WithRuntimeClass sets a deployment's RuntimeClassName
+func WithRuntimeClass(rtc string) DeploymentModifier {
+	return func(d *appsv1.DeploymentArgs) {
+		if podTemplateSpec, ok := ensureDeploymentPodTemplateSpec(d); ok {
+			podTemplateSpec.RuntimeClassName = runtimeClassToPulumi(rtc)
+		}
+	}
+}
+
+// WithServiceAccount sets a deployment's ServiceAccount
+func WithServiceAccount(serviceAccount *corev1.ServiceAccount) DeploymentModifier {
+	return func(d *appsv1.DeploymentArgs) {
+		if podTemplateSpec, ok := ensureDeploymentPodTemplateSpec(d); ok {
+			podTemplateSpec.ServiceAccount = serviceAccount.Metadata.Name()
+		}
+	}
+}
+
+func ensureDeploymentPodMetadata(d *appsv1.DeploymentArgs) (*metav1.ObjectMetaArgs, bool) {
+	// nil check spec
+	specPtr := d.Spec
+	if specPtr == nil {
+		d.Spec = &appsv1.DeploymentSpecArgs{}
+	}
+	// type check spec
+	spec, ok := specPtr.(*appsv1.DeploymentSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check spec.Template
+	templatePtr := spec.Template
+	if templatePtr == nil {
+		spec.Template = &corev1.PodTemplateSpecArgs{}
+	}
+	// type check spec.Template
+	template, ok := templatePtr.(*corev1.PodTemplateSpecArgs)
+	if !ok {
+		return nil, false
+	}
+
+	// nil check template.Metadata
+	metadataPtr := template.Metadata
+	if metadataPtr == nil {
+		template.Metadata = &metav1.ObjectMetaArgs{}
+	}
+
+	metadata, ok := metadataPtr.(*metav1.ObjectMetaArgs)
+	if !ok {
+		return nil, false
+	}
+
+	return metadata, true
+}
+
+// WithLabels appends/ovewrites a Deployment template's labels
+func WithLabels(labels map[string]string) DeploymentModifier {
+	return func(d *appsv1.DeploymentArgs) {
+		if metadata, ok := ensureDeploymentPodMetadata(d); ok {
+			// If labels is nil initialize it
+			if metadata.Labels == nil {
+				metadata.Labels = pulumi.StringMap{}
+			}
+			// merge the existing labels with new ones
+			merged := mergeStringMaps(metadata.Labels.(pulumi.StringMap), rawMapToStringMap(labels))
```

Comment:
Let’s use the standard function instead.

```suggestion
			merged := mergeStringMaps(metadata.Labels.(pulumi.StringMap), pulumi.ToStringMap(labels))
```